packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -192,6 +191,15 @@ export const assertActivableState = (agreement: Agreement): void => {
   }
 };
 
+export const assertRequesterIsDelegate = (
+  delegateId: TenantId | undefined,
+  organizationId: TenantId
+): void => {
+  if (organizationId !== delegateId) {
+    throw operationNotAllowed(organizationId);
+  }
+};
+
```

Comment:
Reuse the same method now on `develop`.
I would remove the export so that it cannot be used generically without referring explicitly to a producer or consumer delegation, so that this is only a helper for other validators in this file. Then you could create an `assertRequesterIsDelegateConsumer` or similar

Reply:
Since I started working on #1162, I did a PR on `develop` to improve the existing validators and make our job simpler, if it makes sense to you we could then do the same but for consumer delegations in our PRs  https://github.com/pagopa/interop-be-monorepo/pull/1246